lib/plausible_web/controllers/auth_controller.ex

```diff
@@ -24,52 +24,61 @@ defmodule PlausibleWeb.AuthController do
               :activate_form
             ]
 
-  def register_form(conn, _params) do
-    if Keyword.fetch!(Application.get_env(:plausible, :selfhost), :disable_registration) != false do
-      redirect(conn, to: Routes.auth_path(conn, :login_form))
-    else
-      changeset = Plausible.Auth.User.changeset(%Plausible.Auth.User{})
+  plug :maybe_disable_registration when action in [:register_form, :register]
 
-      render(conn, "register_form.html",
-        changeset: changeset,
-        layout: {PlausibleWeb.LayoutView, "focus.html"}
-      )
+  defp maybe_disable_registration(conn, _opts) do
+    selfhost_config = Application.get_env(:plausible, :selfhost)
+    disable_registration = Keyword.fetch!(selfhost_config, :disable_registration)
+    first_launch? = Release.first_launch?()
+
+    cond do
+      first_launch? ->
+        conn
```

Comment:
when `first_launch?`, it still shows the registration forms, even with registration disabled, otherwise the user won't be able to get into the system